compiler/gen_OCaml.ml

```diff
@@ -888,7 +888,7 @@ struct
                   ~field:$str:name$
                   e loc
             >>
-        | Some expr ->
```

Comment:
Not sure why there's a separate branch for the case where a default value exists, where deliberately only `Bad_wire_type` errors are caught.